### PR TITLE
Add equality-based dispatchers for testing

### PR DESCRIPTION
**Pull request by radix**
Some really simple dispatchers for testing.

- Give EQDispatcher a map like {MyIntent('foo'): 'bar'} and MyIntent('foo') will be performed to have the result 'bar'.
- Give EQFDispatcher a map like {MyIntent('foo'): lambda i: 'bar'}, and likewise.

@tomprince I'd appreciate your feedback :)

### CODE REVIEW

File: effect/testing.py [hunk 1]
```diff
@@ -164,3 +164,69 @@ def resolve_stubs(dispatcher, effect):
...
+            if k == intent:
+                return sync_performer(lambda d, i: v)
+
+
+class EQFDispatcher(object):
+    """
+    An Equality-based function dispatcher.
+
+    This dispatcher looks up intents by equality and performs them by invoking
+    an associated function.
```
Comment (cyli) [inline effect/testing.py]: Might want to document that the function needs to take a single argument: the intent

**Comment by cyli** [comment]
Non-blocking - Should there be tests for catch?  It is pretty simple, but for full test coverage... :)

**Comment by cyli** [comment]
LGTM on travis +1 :)

**Comment by radix** [comment]
Thank you very much @cyli!

**Pull request merged**